pkg/nvcdi/lib-nvml.go

```diff
@@ -118,10 +118,23 @@ func (l *nvmllib) GetDeviceSpecsBy(identifiers ...device.Identifier) ([]specs.De
 	}
 	defer func() {
 		if r := l.nvmllib.Shutdown(); r != nvml.SUCCESS {
-			l.logger.Warningf("failed to shutdown NVML: %w", r)
+			l.logger.Warningf("failed to shutdown NVML: %v", r)
 		}
 	}()
 
+	if l.nvsandboxutilslib != nil {
```

Comment:
For the time being, I have duplicated the code for `GetAllDeviceSpecs` -- which is exercised more frequently -- and will clean the up more in a post-release follow-up.

Reply:
Would it be worth to create a func now that we have the same code in 2 funcs?

Reply:
I don't know whether a functon would work since we need to defer the calls to `Shutdown()`. It is something we could look into, but I don't want to make that change this close to the release.

Reply:
Agree, something to workshop for the next release. The patch does as described.